src/domain/opendatafabric/src/formats/did_odf.rs

```diff
@@ -139,7 +139,10 @@ impl<'a> DidOdfFmt<'a> {
         let len = {
             let mut c = std::io::Cursor::new(&mut buf[..]);
             write!(c, "{self}").unwrap();
-            c.position() as usize
+            #[cfg_attr(target_pointer_width = "64", allow(clippy::cast_possible_truncation))]
+            {
+                c.position() as usize
+            }
```

Comment:
A bad one. File position is `u64` even on 32-bit architectures. On the other hand, the buffer must be of addressable size. I think here we need a dynamically checked cast that would panic otherwise. Maybe the one from this `conv` library.

Reply:
used `usize::try_from()` -- done